src/jumio-kyc/kyc.service.ts

```diff
@@ -304,4 +308,44 @@ export class KycService {
 
     return signature === expectedSignature;
   }
+
+  async allocate(pool: Pool, redemptions: Redemption[]): Promise<void> {
+    //pool_one -> pool1_points
+    const pointsColumn = this.userPointsService.poolToColumn(pool);
+    const airdropPool = AIRDROP_CONFIG.data.find((c) => c.name === pool);
+    assert.ok(airdropPool);
+    const poolOre = airdropPool.coins * ORE_TO_IRON;
+
+    // find eligble users, filter -> passed kyc
+    const totalPoints = await this.userPointsService.poolTotal(
+      pool,
+      redemptions.map((r) => r.user_id),
+    );
+
+    // exit early if allocations already made, wipe column and rerun if needed
+    const alreadyAllocated = redemptions.some((r) => r[pool]);
+    if (alreadyAllocated) {
+      return;
+    }
+
+    for (const redemption of redemptions) {
+      const userPoints = await this.userPointsService.findOrThrow(
+        redemption.user_id,
+      );
+      const userPoolPoints = userPoints[pointsColumn];
+      if (!userPoolPoints) {
+        continue;
+      }
+
+      const update: {
+        pool_one?: number;
+        pool_two?: number;
+        pool_three?: number;
+        pool_four?: number;
+      } = {};
+      const allocatedOre = Math.floor((userPoolPoints / totalPoints) * poolOre);
+      update[pool] = allocatedOre;
```

Comment:
you can just inline this and do this, and delete all the other code above like the type

{
 [pool]: alloactedOre
}